lib/ex_doc/autolink.ex

```diff
@@ -383,21 +392,40 @@ defmodule ExDoc.Autolink do
 
   ## Internals
 
-  defp module_url(module, config) do
-    if module == config.current_module do
-      "#content"
-    else
-      if Refs.public?({:module, module}) do
-        module_url(tool(module), module, config)
-      end
+  defp module_url(module, module_string, mode, config, mix_task? \\ false)
+
+  defp module_url(module, _module_string, _mode, %{current_module: module}, _mix_task?),
+    do: "#content"
+
+  defp module_url(module, module_string, mode, config, mix_task?) do
+    case {mode, Refs.get_visibility({:module, module})} do
+      {_, :public} ->
+        module_url_app(tool(module), module, config)
+
+      {:regular_link, :undefined} ->
+        nil
+
+      {_, visibility} ->
+        module_string =
+          if mix_task? do
```

Comment:
since we already have an if statement here, maybe let’s make the warning specific for mix tasks here? “(...) references module mix foo (...)” isn’t totally accurate.

Reply:
how about this message?